pkg/sfu/downtrack.go

```diff
@@ -1675,6 +1711,52 @@ func (d *DownTrack) HandleRTCPSenderReportData(_payloadType webrtc.PayloadType,
 	return nil
 }
 
+func (d *DownTrack) SetUpstreamRTT(rttMs uint32) {
+	if d.upstreamRtt.Load() == rttMs {
+		return
+	}
+
+	d.upstreamRtt.Store(rttMs)
+	d.updatePlayoutDelay()
+}
+
+func (d *DownTrack) updatePlayoutDelay() {
+	if d.playoutDelayExtID == 0 {
+		return
+	}
+
+	rttMax := d.upstreamRtt.Load()
+	if rtt := d.rtpStats.GetRtt(); rtt > rttMax {
+		rttMax = rtt
```

Comment:
Can we add an even more smoothed version of RTT? (looks like it is doing average of current and new now). RTT can get quite high some times. Transient issues affecting a few packets could affect the RTCP receiver report or sender report causing RTT to be high. I am thinking we do not want that to update play out delay? Play out delay gets applied to all frames right? Also, when we change play out delay does client slowly move from one value to next or does it jump?

Reply:
Probably worth thinking about quantising RTT too. Maybe, in buckets of 50 ms. With video, there could be several updates per second (RTCP Receiver reports received several times a second). Just want to make sure that it does not affect smoothness.

Reply:
The average is a smooth change, newTarget = current + (target-current)/2. I will test if we need a more smooth factor to resist the sudden jitter of network change. Since we have [min,max] limit of the delay, it would be ok to filter an immediately very high RTT. It sounds good to add a filter for several rtt updates per second.